crates/bpe/src/byte_pair_encoding.rs

```diff
@@ -192,7 +205,9 @@ impl BytePairEncoding {
         let mut token_starts = vec![0];
         let mut bytes_hash_to_token = FnvHashMap::default();
         for (i, token) in iter.enumerate() {
-            bytes_hash_to_token.insert(hash_bytes(&token), i as u32);
+            if let Some(j) = bytes_hash_to_token.insert(hash_bytes(&token), i as u32) {
+                eprintln!("collision: ({i}, {j})");
+            }
```

Comment:
should the function return an error instead when this happens?

Reply:
If this happens, the assertion below the loop will fail. I wonder if we want to provide a way to specify the factor as part of the API, if anyone wants to construct a BPE from their own dictionary. The hard-coded constant might make it harder to reuse this if users bring their own tokens. But I'll save that for a follow-up PR.